src/scene/materials/material.js

```diff
@@ -393,11 +404,16 @@ class Material {
     }
 
     clearVariants() {
+
+        // clear variants on the material
         this.variants = {};
+
+        // but also clear them from all materials that reference them
         for (let i = 0; i < this.meshInstances.length; i++) {
             const meshInstance = this.meshInstances[i];
-            for (let j = 0; j < meshInstance._shader.length; j++) {
-                meshInstance._shader[j] = null;
+            const shaders = meshInstance._shader;
+            for (let j = 0; j < shaders.length; j++) {
```

Comment:
This looks like filling the `shaders` array with `null`'s, why not just `shaders.length = 0`, is it required to keep the length of the array for maybe performance reasons?